lib/MusicBrainz/Server/Data/Release.pm

```diff
@@ -1004,31 +1004,34 @@ sub merge
     );
 
     if ($merge_strategy == $MERGE_APPEND) {
-        my %positions = %{ $opts{medium_positions} || {} }
+        my %positions_by_medium = %{ $opts{medium_positions} || {} }
             or confess('Missing medium_positions parameter');
 
+        my %new_positions = map { $_ => 1 } values %positions_by_medium;
```

Comment:
Creates a "set" of all the new positions, so we can easily check if a position is used.